ServerSocketTest/src/main/Main.java

```diff
@@ -50,38 +74,123 @@ public static void main(String[] args) {
+				*/
+				if(method.equals("GET"))
 				{
-					OutputStream clientOutput=client.getOutputStream();//output all responses
-					
-					clientOutput.write(("HTTP/1.1 200 OK\r\n").getBytes());//encode to bytes
-					clientOutput.write(("\r\n").getBytes());//blank line
-					clientOutput.write(("Mark \r\nDorian \r\nMehir \r\nDan \r\nKhate \r\nRobert \r\nCiaran \r\nDillon").getBytes());//encode to bytes
+					if(resource.equals("/people"))//check the resource we are looking for
+					{
+						OutputStream clientOutput=client.getOutputStream();//output all responses
+						
+						clientOutput.write(("HTTP/1.1 200 OK\r\n").getBytes());//encode to bytes
+						clientOutput.write(("\r\n").getBytes());//blank line
+						//people are contained in an arraylist. So when queried we output everyone in the list
+						for(int i=0;i<people.size();i++)
+						{
+							clientOutput.write((people.get(i)+" \r\n").getBytes());//encode to bytes
+							
+						}
 				
-					clientOutput.flush();//empty the built up buffer
+						
+						clientOutput.flush();//empty the built up buffer
+					}
+					else if(resource.equals("/"))//the root doesnt do anything rn. Just a "welcome page"
+					{
+						OutputStream clientOutput=client.getOutputStream();//output all responses
+						
+						clientOutput.write(("HTTP/1.1 200 OK\r\n").getBytes());//encode to bytes
+						clientOutput.write(("\r\n").getBytes());//blank line
+						clientOutput.write(("Hello").getBytes());//encode to bytes
+						
+						clientOutput.flush();//empty the built up buffer
+						
+					}
+					else {
+						//if the resource is not found output 404
+						OutputStream clientOutput=client.getOutputStream();//output all responses
+						clientOutput.write(("HTTP/1.1 404 Not Found\r\n").getBytes());//encode to bytes
+						clientOutput.write(("\r\n").getBytes());//blank line
+						clientOutput.write(("404").getBytes());//encode to bytes
+						clientOutput.flush();//empty the built up buffer
+					}
+					
 				}
-				else
+				/*if the method is post we need to get the info stored in the body. The body comes after the header. W
+				 * we need to check the content length to know when we are done and we need to check content type so we know 
+				 * which parser to use. currently only Content-Type: application/x-www-form-urlencoded . is supported but need to look into this more*/
+				else if(method.equals("POST"))
 				{
-					OutputStream clientOutput=client.getOutputStream();//output all responses
-					
-					clientOutput.write(("HTTP/1.1 200 OK\r\n").getBytes());//encode to bytes
-					clientOutput.write(("\r\n").getBytes());//blank line
-					clientOutput.write(("Hello").getBytes());//encode to bytes
+									
+					if(resource.equals("/people"))
+					{
+						int contentLength=0;
+						//parse out content length using the much beloved regex
+						   // Define the pattern to match "Content-Length: value"
+				        Pattern pattern = Pattern.compile("Content-Length: (\\d+)");
```

Comment:
I'm confused about the whole pattern stuff.

Reply:
Content-Length is apart of the header of a request and an example of a header from a post request is:
POST /people HTTP/1.1
Host: localhost:8097
User-Agent: curl/8.4.0
Accept: */*
Content-Length: 65
Content-Type: application/x-www-form-urlencoded


We need to parse out the Content-Length to find the value 65 (headers can have many lines above is just an example). There are many ways to parse this information. One of these ways is to search the header for a pattern. I define this pattern using the built in pattern class and regular expression .

So line 140 means search the header until you reach the line containing "Content-Length" once we find this line extract the data contained within the brackets () which will be a number of various length. I need this number so I know how much info is in the body of the request.